### PR TITLE
Add caching for certificates, makes things much faster.

### PR DESCRIPTION
**Pull request by ukd1**
@bmizerany @kr this is my attempt at adding caching of the certificates to mitm. Thoughts?

### CODE REVIEW

File: mitm.go [hunk 1]
```diff
@@ -115,7 +116,27 @@ func (p *Proxy) serveConnect(w http.ResponseWriter, r *http.Request) {
 }
 
 func (p *Proxy) cert(names ...string) (*tls.Certificate, error) {
-	return genCert(p.CA, names)
+	var (
+		cache = map[string]tls.Certificate{} // names (joined with ,) -> cert
```
Comment (kr) [inline mitm.go]: It looks like the cache is in a local variable and thrown away when this method returns.

Comment (ukd1) [inline mitm.go]: Ok; weird as it feels quicker (placebo?), also I borrowed the concept from some code of @bmizerany's. 

I'll read up on how to do this properly, if you're down for using it? :+1: 

(btw, I'm a beginner at go, so plz forgive me!)

Comment (kr) [inline mitm.go]: Sure. In principle, caching the generated cert is totally fine. Just be prepared to show some measurements of how much it helps speed and what the tradeoff would be in increased memory use. :) For example, before and after results of a benchmark function would be excellent.

Comment (ukd1) [inline mitm.go]: @kr ok, will do!

**Pull request closed without merging**